docs/stream-exporter-design.md

```diff
@@ -0,0 +1,207 @@
+# StreamExporter Design
+
+In strongly typed languages typically there will be 2 separate `Exporter` interfaces, one that accepts spans from a tracer (SpanExporter) and one that accepts metrics (MetricsExporter)
+
+The exporter SHOULD be called with a checkpoint of finished (possibly dimensionally reduced) export records. Most configuration decisions have been made before the exporter is invoked, including which instruments are enabled, which concrete aggregator types to use, and which dimensions to aggregate by.
+
+## Use Cases
+
+Monitoring and alerting systems commonly use the data provided through metric events or tracers, after applying various aggregations and converting into various exposition format. After getting the data, the systems need to be able to see the data. The StreamExporter will be used here to print data through an ostream, this is seen as a simple exporter where the user doesn’t have the burden of implementing or setting up a protocol dependent exporter.
+
+The StreamExporter will also be used as a debugging tool for the Metrics API/SDK and Tracing API/SDK which are currently work in progress projects. This exporter will allow contributors to easily diagnose problems when working on the project.
+
+## Push vs Pull Exporter
+
+There are two different versions of exporters: Push and Pull. A Push Exporter pushes the data outwards towards a system, in the case of the StreamExporter it sends its data into an ostream. A Pull Exporter exposes data to some endpoint for another system to grab the data. 
+
+The StreamExporter will only be implementing a Push Exporter framework.
+
+## Design Tenets
+
+* Reliability
+    * The Exporter should be reliable; data exported should always be accounted for. The data will either all be successfully exported to the destination server, or in the case of failure, the data is dropped. `Export` will always return failure or success to notify the user of the result.
+    * Thread Safety
+        * The StreamExporter can be called simultaneously, however we do not handle this in the Exporter. Synchronization should be done at a lower level.
```

Comment:
I changed/moved our thread safety to Design Tenets, does this cover what you were saying about simultaneous calls? @reyang